packages/core/src/utils/file-utils.ts

```diff
@@ -78,18 +77,44 @@ export function symlinkJunction(to: string, from: string) {
  * returns the first occurrence of the node_modules, or null of none could be found.
  * @param basePath starting point
  */
-export async function findNodeModules(basePath: string): Promise<string | null> {
+export async function findNodeModulesList(
+  basePath: string
+): Promise<{
+  nodeModulesList: string[];
+  basePath: string;
+}> {
   basePath = path.resolve(basePath);
-  const nodeModules = path.resolve(basePath, 'node_modules');
-  try {
-    await fs.promises.stat(nodeModules);
-    return nodeModules;
-  } catch (e) {
-    const parent = path.dirname(basePath);
-    if (parent === basePath) {
-      return null;
-    } else {
-      return findNodeModules(path.dirname(basePath));
+  const nodeModulesList: string[] = [];
+
+  const dirBfsQueue: string[] = (await fs.promises.readdir(basePath)) ?? [];
```

Comment:
I think we should make sure that the `node_modules` is a directory and that it is not already a symlink. We can probably use `withFileTypes: true` to get this information all in one go:

https://nodejs.org/api/fs.html#fs_fspromises_readdir_path_options

Reply:
Yes, thank you for notice me. infact, readdir on file cause the error on e2e test, this is fixed now